tests/unit/src/test/scala/tests/debug/BreakpointDapSuite.scala

```diff
@@ -0,0 +1,468 @@
+                |    val bar = new Bar
+                |    bar.call()
+                |  }
+                |}
+                |""".stripMargin
+  )
+
+  assertFileBreakpoints("case-class")(
+    source = """|a/src/main/scala/a/Main.scala
+                |package a
+                |
+                |case class Bar() {
+                |  def call() = {
+                |>>  println()
+                |  }
+                |}
+                |
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    val bar = new Bar
+                |    bar.call()
+                |  }
+                |}
+                |""".stripMargin
+  )
+
+  assertFileBreakpoints("case-class-unapply")(
+    source = """|a/src/main/scala/a/Main.scala
+                |package a
+                |
+                |case class Bar() {
+                |  def unapply(arg: Any): Option[Int] = {
+                |>>    Some(1)
+                |  }
+                |}
+                |
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    val bar = Bar()
+                |    this match {
+                |      case bar(1) => println()
+                |      case _ =>
+                |    }
+                |  }
+                |}
+
+                |""".stripMargin
+  )
+
+  assertFileBreakpoints("companion")(
+    source = """|a/src/main/scala/a/Main.scala
+                |package a
+                |
+                |case class Bar()
+                |
+                |object Bar {
+                |  def call() = {
+                |>>  println()
+                |  }
+                |}
+                |
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    Bar.call()
+                |  }
+                |}
+                |""".stripMargin
+  )
+
+  assertFileBreakpoints("companion-apply")(
+    source = """|a/src/main/scala/a/Main.scala
+                |package a
+                |
+                |case class Bar()
+                |
+                |object Bar {
+                |  def apply() = {
+                |>>  println()
+                |  }
+                |}
+                |
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    Bar()
+                |  }
+                |}
+                |""".stripMargin
+  )
+
+  assertFileBreakpoints("for-comprehension")(
+    source = """|a/src/main/scala/a/Main.scala
+                |package a
+                |
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    for {
+                |>>    x <- List()
+                |    } println(x)
+                |  }
+                |}
+                |""".stripMargin
+  )
+
+  // TODO investigate what happens
```

Comment:
Let's open followup ticket

Reply:
Did we open a followup ticket?

Reply:
Nope, I was iterating over a `List()` which is empty, hence the breakpoint was not effective